src/join_mesh.py

```diff
@@ -1,140 +1,254 @@
 #!/usr/bin/env python3
+"""
+Joins meshes partitioned by ASTE mesh partitioner.
+
+There is two possible way of merging:
+- Partition-wise (does not recovery original mesh order and lack of discarded cells)
+- Recovery merge (maintain original mesh order and adds discarded cells)
+
+Sample usage 
+
+partitioned files:
+prefix_<#>.vtu
+
+./join_mesh.py prefix -o joined_mesh.vtk
+
+./join_mesh.py prefix -o joined_mesh.vtk -r ./recovery/path/
+
+"""
+
 import argparse
 import logging
 import os
-import numpy as np
 import json
-from mesh_io import read_txt, get_cell_type, read_conn
-import mesh_io
+import vtk
 
+def parse_args():
+    parser = argparse.ArgumentParser(description="Read a partitioned mesh and join it into a .vtk or .vtu file.")
+    parser.add_argument("in_meshname", metavar="inputmesh", help="The partitioned mesh prefix used as input")
```

Comment:
The prefix I took is like that
`prefix_1.vtu
prefix_2.vtu
prefix_3.vtu 
...` 

<prefix>_<#filerank>.<extension> 

Since both preCICE and ASTE produce `.vtu` extension I just only look for `.vtu` files.

Reply:
Ah yes, it would be nice to have your naming convention in the help  message `[prefix]_[#filerank].[extension]`